Tests/SonarScanner.MSBuild.Test/AnalysisWarning/WarningsSerializerTest.cs

```diff
@@ -0,0 +1,67 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarScanner.MSBuild.AnalysisWarning;
+
+namespace SonarScanner.MSBuild.Test.AnalysisWarning
+{
+    [TestClass]
+    public class WarningsSerializerTest
+    {
+        public TestContext TestContext { get; set; }
+
+        [TestMethod]
+        public void SerializeToFile_InvalidWarningsCollections_Throws()
+        {
+            Action act = () => WarningsSerializer.Serialize(null, "filePath");
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("warnings");
+        }
+
+        [TestMethod]
+        public void SerializeToFile_InvalidFileDirectory_Throws()
+        {
+            Action act = () => WarningsSerializer.Serialize(Array.Empty<Warning>(), null);
+            act.Should().Throw<ArgumentException>().And.ParamName.Should().Be("filePath");
+        }
+
+        [TestMethod]
```

Comment:
could you please add tests with:
- an empty `Warning[]` array and a valid path?
- a valid array and an invalid path (e.g. inexistant folder or folder with no write permissions)